tests/schema_registry/test_avro_serializer.py

```diff
@@ -74,6 +79,62 @@ def test_avro_serializer_config_auto_register_schemas_false(mock_schema_registry
     assert test_client.counter['POST'].get('/subjects/{}'.format(subject)) == 1
 
 
+def test_avro_serializer_multiple_topic_per_serializer_instance(mock_schema_registry):
+    """
+    Ensures schema_id is correctly find when same serializer is used for multiple topics
+    """
+    conf = {'url': TEST_URL}
+    test_client = mock_schema_registry(conf)
+    topic1 = "test-topic1"
+    topic2 = "test-topic2"
+
+    test_serializer = AvroSerializer("string", test_client,
+                                     conf={'auto.register.schemas': False})
+
+    def ensure_id_match(ctx):
+        subject = "{}-{}".format(ctx.topic, ctx.field)
+        expected_id = find_schema_id(subject)
+
+        payload = test_serializer("test", ctx)
+        _, schema_id = unpack('>bI', BytesIO(payload).read(5))
+        assert schema_id == expected_id
+
+    ensure_id_match(SerializationContext(topic1, MessageField.KEY))
+    ensure_id_match(SerializationContext(topic2, MessageField.VALUE))
+    ensure_id_match(SerializationContext(topic1, MessageField.KEY))
+
+    # Ensure lookup_schema was invoked only once per shema
+    assert test_client.counter['POST'].get('/subjects/{}-key'.format(topic1)) == 1
+    assert test_client.counter['POST'].get('/subjects/{}-value'.format(topic2)) == 1
+
+
+def test_avro_serializer_preload_schema_id(mock_schema_registry):
+    """
+    Ensures serializer do not reload schema ID from registry after user has force its preloading.
+    """
+    conf = {'url': TEST_URL}
+    test_client = mock_schema_registry(conf)
+    topic1 = "test-topic1"
+    topic2 = "test-topic2"
+
+    test_serializer = AvroSerializer("string", test_client,
+                                     conf={'auto.register.schemas': False})
+
+    test_serializer.load_registry_schema_id(SerializationContext(topic1, MessageField.KEY))
```

Comment:
i haven't looked at this closely, but given the existing design at least, the capability to pre-cache a schema seems more like it belongs in the (caching) schema registry client, not the serializer.

Also, the `SerializationContext` is intended to be used for what it's name implies - context whilst (de)serializing, so it shouldn't be part of the interface otherwise.

Reply:
Looking at your comment, indeed the `load_registry_schema_id` should not be part of the interface.

I have look more deeply into the `SchemaRegistryClient`.
There is indeed a cache into the `register_schema` function used when `'auto.register.schemas': True`.
But there is not in `lookup_schema` function used when `'auto.register.schemas': False`.

This is where my miss understanding comes from.

---

To go forward:

1) I can update my confluent_kafka usage to `register_schema` and force pre-loading.
2) If you are still interested by the others changes: I can revert my last commit (related to making public the usage of `load_registry_schema_id`). Otherwise you can close the PR.
3) It may still be interesting to add a cache into the `lookup_schema` and remove the one from the `Serializer` object.